src/server.py

```diff
@@ -34,22 +38,29 @@ def server():
                 message_complete = True
 
         try:
-            parse_request(recv_message.decode('utf-8'))
+            client_uri = parse_request(recv_message.decode('utf-8'))
         except TypeError as msg:
             reply = response_error(405, str(msg))
         except ValueError as msg:
             reply = response_error(400, str(msg))
         else:
```

Comment:
Interesting -- I haven't used the `try except else` pattern much, but I like what you're doing here.